lib/Maskings/Maskings.cpp

```diff
@@ -0,0 +1,309 @@
+    std::string msg = "maskings file '" + filename + "' is empty";
+    LOG_TOPIC(DEBUG, Logger::CONFIG) << msg;
+    return MaskingsResult(MaskingsResult::CANNOT_READ_FILE, msg);
+  }
+
+  std::unique_ptr<Maskings> maskings(new Maskings{});
+
+  try {
+    std::shared_ptr<VPackBuilder> parsed =
+        velocypack::Parser::fromJson(definition);
+
+    ParseResult<Maskings> res = maskings->parse(parsed->slice());
+
+    if (res.status != ParseResult<Maskings>::VALID) {
+      return MaskingsResult(MaskingsResult::ILLEGAL_DEFINITION, res.message);
+    }
+
+    return MaskingsResult(std::move(maskings));
+  } catch (velocypack::Exception const& e) {
+    std::string msg =
+        "cannot parse maskings file '" + filename + "': " + e.what();
+    LOG_TOPIC(DEBUG, Logger::CONFIG) << msg << ". file content: " << definition;
+
+    return MaskingsResult(MaskingsResult::CANNOT_PARSE_FILE, msg);
+  }
+}
+
+ParseResult<Maskings> Maskings::parse(VPackSlice const& def) {
+  if (!def.isObject()) {
+    return ParseResult<Maskings>(ParseResult<Maskings>::DUPLICATE_COLLECTION,
+                                 "expecting an object for masking definition");
+  }
+
+  for (auto const& entry : VPackObjectIterator(def, false)) {
+    std::string key = entry.key.copyString();
+    LOG_TOPIC(TRACE, Logger::CONFIG) << "masking collection '" << key << "'";
+
+    if (_collections.find(key) != _collections.end()) {
+      return ParseResult<Maskings>(ParseResult<Maskings>::DUPLICATE_COLLECTION,
+                                   "duplicate collection entry '" + key + "'");
+    }
+
+    ParseResult<Collection> c = Collection::parse(entry.value);
+
+    if (c.status != ParseResult<Collection>::VALID) {
+      return ParseResult<Maskings>(
+          (ParseResult<Maskings>::StatusCode)(int)c.status, c.message);
+    }
+
+    _collections[key] = c.result;
+  }
+
+  return ParseResult<Maskings>(ParseResult<Maskings>::VALID);
+}
+
+bool Maskings::shouldDumpStructure(std::string const& name) {
+  auto const itr = _collections.find(name);
+
+  if (itr == _collections.end()) {
+    return false;
+  }
+
+  switch (itr->second.selection()) {
+    case CollectionSelection::FULL:
+      return true;
+    case CollectionSelection::IGNORE:
+      return false;
+    case CollectionSelection::STRUCTURE:
+      return true;
+  }
+}
+
+bool Maskings::shouldDumpData(std::string const& name) {
+  auto const itr = _collections.find(name);
+
+  if (itr == _collections.end()) {
+    return false;
+  }
+
+  switch (itr->second.selection()) {
+    case CollectionSelection::FULL:
+      return true;
+    case CollectionSelection::IGNORE:
+      return false;
+    case CollectionSelection::STRUCTURE:
+      return false;
+  }
+}
+
+VPackValue Maskings::maskedItem(Collection& collection,
+                                std::vector<std::string>& path,
+                                std::string& buffer, VPackSlice const& data) {
+  static std::string xxxx("xxxx");
```

Comment:
Should be declared outside of this function, so we can avoid rechecking if the static was already initialized or not.